lib/checkbufferoverrun.cpp

```diff
@@ -1069,3 +1069,67 @@ void CheckBufferOverrun::objectIndexError(const Token *tok, const ValueFlow::Val
                 CWE758,
                 Certainty::normal);
 }
+
+static bool isVLAIndex(const Token* tok)
+{
+    if (!tok)
+        return false;
+    if (tok->varId() != 0U)
+        return true;
+    if (tok->str() == "?") {
+        // this is a VLA index if both expressions around the ":" is VLA index
```

Comment:
isn't it `if either expression is a VLA index `..

    buf[x ? y : 10]  // <- this is a VLA right?

Reply:
That code snippet has a VLA, yes.

Actually, the comment is not mine as I just restored a check that someone removed. In general, you are right, but this is not what the check below (line 1101) wants: "// don't warn about constant negative index because that is a compiler error"
Thus, it checks that both sides of the ternary : is variable to ensure that the negative values looked for in line 1100 are not from a constant.

Reply:
yes hmm.. the code is not perfect but it does not seem trivial to make it perfect neither.